pkg/controllers/state/cluster.go

```diff
@@ -419,6 +422,15 @@ func (c *Cluster) MarkPodSchedulingDecisions(ctx context.Context, podErrors map[
 			}
 		}
 	}
+	c.UpdatePodToNodeClaimMapping(ncPods)
+}
+
+func (c *Cluster) UpdatePodToNodeClaimMapping(ncPods map[string][]*corev1.Pod) {
+	for ncName, pods := range ncPods {
+		for _, p := range pods {
+			c.podToNodeClaim.LoadOrStore(client.ObjectKeyFromObject(p), ncName)
```

Comment:
I don't think you want to LoadOrStore -- I think you want to just Store. If the mapping to the NodeClaim changes, I think we want to know it since this affects our metrics

Reply:
Agreed and updated.